Fabric.Authorization.API/scripts/Install-Authorization-Windows.ps1

```diff
@@ -29,94 +44,268 @@ function Invoke-MonitorShallow($authorizationUrl)
+    $allCerts |
+        ForEach-Object {New-Object PSCustomObject -Property @{
+        'Index'=$index;
+        'Subject'= $_.Subject; 
+        'Name' = $_.FriendlyName; 
+        'Thumbprint' = $_.Thumbprint; 
+        'Expiration' = $_.NotAfter
+        };
+        $index ++} |
+        Format-Table Index,Name,Subject,Expiration,Thumbprint  -AutoSize
+
+    $selectionNumber = Read-Host  "Select a signing and encryption certificate by Index"
+    Write-Host ""
+    if([string]::IsNullOrEmpty($selectionNumber)){
+		Write-Error "You must select a certificate so Fabric.Identity can sign access and identity tokens." -ErrorAction Stop
+    }
+    $selectionNumberAsInt = [convert]::ToInt32($selectionNumber, 10)
+    if(($selectionNumberAsInt -gt  $allCerts.Count) -or ($selectionNumberAsInt -le 0)){
+        Write-Error "Please select a certificate with index between 1 and $($allCerts.Count)."  -ErrorAction Stop
+    }
+    $certThumbprint = Get-CertThumbprint $allCerts $selectionNumberAsInt     
+    $primarySigningCertificateThumbprint = $certThumbprint -replace '[^a-zA-Z0-9]', ''    
+    $encryptionCertificateThumbprint = $certThumbprint -replace '[^a-zA-Z0-9]', ''
+    }catch{
+        $scriptDirectory =  Get-CurrentScriptDirectory
+        Set-Location $scriptDirectory
+        Write-Error "Could not set the certificate thumbprint. Error $($_.Exception.Message)" -ErrorAction Stop        
 }
 
 try{
-	$accessToken = Get-AccessToken -authUrl $identityServerUrl -clientId "fabric-installer" -scope "fabric/identity.manageresources" -secret $fabricInstallerSecret
-} catch {
-	Write-Host "There was a problem getting an access token for the Fabric Installer client, please make sure that Fabric.Identity is running and that the fabricInstallerSecret value in the install.config is correct. Halting installation."
+    $signingCert = Get-Certificate $primarySigningCertificateThumbprint
+}catch{
+    Write-Error "Could not get signing certificate with thumbprint $primarySigningCertificateThumbprint. Please verify that the primarySigningCertificateThumbprint setting in install.config contains a valid thumbprint for a certificate in the Local Machine Personal store."
+    throw $_.Exception
+}
+
+try{
+	$encryptionCert = Get-Certificate $encryptionCertificateThumbprint
+}catch{
+	Write-Host "Could not get encryption certificate with thumbprint $encryptionCertificateThumbprint. Please verify that the encryptionCertificateThumbprint setting in install.config contains a valid thumbprint for a certificate in the Local Machine Personal store. Halting installation."
 	throw $_.Exception
-	exit 1
 }
 
+
+$userEnteredFabricInstallerSecret = Read-Host  "Enter the Fabric Installer Secret or hit enter to accept the default [$fabricInstallerSecret]"
+Write-Host ""
+if(![string]::IsNullOrEmpty($userEnteredFabricInstallerSecret)){   
+     $fabricInstallerSecret = $userEnteredFabricInstallerSecret
+}
+<# if([string]::IsNullOrEmpty($fabricInstallerSecret -eq $null)){
```

Comment:
Do we still need this code? It looks to be commented out.

Reply:
Yeah I can pull that section out.